app/views/tasks/index.html.haml

```diff
@@ -1,23 +1,39 @@
-%h1 Tasks
+- if params.fetch(:filter, nil).present?
```

Comment:
do we actually need the `fetch` here?

Reply:
You could always do `params[:filter].present?` but then it wouldn't default it to `nil`.

Reply:
functionally, they are the same range of results, no?

Reply:
Not entirely.

> hash[:key] || "value" is still faster than #fetch to set default key in Ruby 2.6.  (which is something i just now learned)
> The only reason you want to use `#fetch` is for when you want your code to raise KeyError when a key is not found.

Source: https://sikac.hu/fetch-vs-bracket-or-for-default-value-in-ruby-9bfd3b4cc37e

Reply:
I guess what I had in mind was that we don't actually need a default value for anything. FWIW, I was just curious, this is a pretty minute detail. In any case, this might make more clear what I was thinking...

```ruby
result_1 = params[:filter] # => nil
result_2 = params[:filter] # => 'SomeString'

result_3 = params.fetch(:filter, nil) # => nil
result_4 = params.fetch(:filter, nil) # => 'SomeString'
```
Using the `[]` syntax to return the value of a given key from a hash (or hash-like object) will already return `nil` if it doesn't exist, so the fetch and default `nil` are redundant.